02_knight_and_princess/tamara_gadieva/graph_func.cpp

```diff
@@ -0,0 +1,43 @@
+#include "graph_generate.hpp"
+
+void Graph::add_vertex() {
+  vertices_.push_back(Vertex(get_new_vertex_id()));
+}
+
+void Graph::add_vertex_inf(const VertexId& cur_vertex_id) {
+  vertices_[cur_vertex_id].add_edge_id(get_cur_edge_id());
+}
+
+void Graph::add_edge(const VertexId& from_vertex_id,
+                     const VertexId to_vertex_id) {
+  edges_.push_back(Edge(get_cur_edge_id(), from_vertex_id, to_vertex_id));
+  increase_edge_id();
```

Comment:
1. Зачем разделять на `get_cur_edge_id()` и `increase_edge_id()`? Логика должна быть аналогичной `vertex`
2. После вызова функции, у тебя. граф останется в сломанном, незавершенном состоянии, тебе нужно обязательно обновить вершины внутри этой фунцкции.
3. Ты не можешь забирать вершины и грани из массива по их `id`. У нас нет гарантии, что их `id` совпадают с нумерацией. Что будет, если мы в будущем, например, добавим логику по удалению вершин и граней? Наша программа сломается. Нужно непосредственно находить их внутри коллекций.
4. `push` vs `emplace`
```suggestion
  const auto& new_edge = edges_.emplace_back(get_new_edge_id(), from_vertex_id, to_vertex_id));
  auto& from_vertex = get_vertex(from_vertex_id);
  auto& to_vertex = get_vertex(to_vertex_id);
  from_vertex.add_edge_id(new_edge.id);
  to_vertex.add_edge_id(new_edge.id);
```